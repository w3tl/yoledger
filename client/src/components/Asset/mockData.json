[{
  "__typename": "Account",
  "id": "1",
  "name": "Bank Card",
  "balance": 0
}, {
  "__typename": "Account",
  "id": "2",
  "name": "Cash",
  "balance": 10
}, {
  "__typename": "Account",
  "id": "3",
  "name": "Bank Card 2",
  "balance": 150
}]
