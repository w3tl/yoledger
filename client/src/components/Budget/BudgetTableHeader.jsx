import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Button } from '../Elements';
import widivData from './BudgetTableHeaderHOC';
// eslint-disable-next-line no-unused-vars
function BudgetTableHeader({ periods, onNext, onPrev }) {
  return (
    <div
      style={{
        display: 'flex',
      }}
    >
      <div style={{ flex: '1 100px' }}>
        Account
      </div>
      {periods.map(date => (
        <div key={date} style={{ flex: 3 }}>
          {moment(date).format('D MMM')}
        </div>
      ))}
      <div style={{ flex: 1 }}>
        <Button id="nextButton" onClick={() => onNext(periods[1])}>Next</Button>
      </div>
    </div>
  );
}

BudgetTableHeader.propTypes = {
  periods: PropTypes.arrayOf(PropTypes.string),
  onNext: PropTypes.func,
  onPrev: PropTypes.func,
};

BudgetTableHeader.defaultProps = {
  periods: [],
  onNext: () => {},
  onPrev: () => {},
};

export { BudgetTableHeader as BudgetTableHeaderComponent };
export default widivData(BudgetTableHeader);
