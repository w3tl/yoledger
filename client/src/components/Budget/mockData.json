{
  "__typename": "BudgetPlan",
  "periods": ["2018-06-10", "2018-06-25", "2018-07-10", "2018-07-25", "2018-08-10", "2018-08-25"],
  "accounts": [{
    "account": {
      "__typename": "Account",
      "id": "1",
      "name": "Food"
    },
    "allocations": [{
      "__typename": "Allocation",
      "date": "2018-06-10",
      "amount": 10,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-06-25",
      "amount": 12,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-10",
      "amount": 14,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-25",
      "amount": 18,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-10",
      "amount": 7,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-25",
      "amount": 6,
      "balance": 0
    }]
  }]
}
