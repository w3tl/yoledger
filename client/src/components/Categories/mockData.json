{
  "expenses": [{
    "id": "21",
    "name": "Food",
    "type": "EXPENSE"
  }, {
    "id": "22",
    "name": "Beer",
    "type": "EXPENSE"
  }, {
    "id": "23",
    "name": "Train",
    "type": "EXPENSE"
  }],
  "incomes": [{
    "id": "31",
    "name": "Salary",
    "type": "INCOME"
  }, {
    "id": "32",
    "name": "Debts",
    "type": "INCOME"
  }, {
    "id": "33",
    "name": "Cashback",
    "type": "INCOME"
  }]
}
