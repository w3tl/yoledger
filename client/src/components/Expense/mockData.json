[{
  "id": "21",
  "name": "Food"
}, {
  "id": "22",
  "name": "Beer"
}, {
  "id": "23",
  "name": "Train"
}]
