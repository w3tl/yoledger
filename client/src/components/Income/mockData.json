[{
  "id": "31",
  "name": "Salary"
}, {
  "id": "32",
  "name": "Debts"
}, {
  "id": "33",
  "name": "Dividends"
}]
