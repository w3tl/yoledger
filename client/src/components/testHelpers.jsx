import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

export const wait = (ms = 0) => new Promise(resolve => setTimeout(resolve, ms));

export const withProvider = Component => props => (
  <MockedProvider {...props}>
    <Component />
  </MockedProvider>
);

export const testRenderWithoutError = (Component) => {
  it('renders without error', async () => {
    const wrapper = shallow(Component);
    await wait();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
};

export const testLoadingState = (Component) => {
  it('should render loading state initially', () => {
    const wrapper = mount(Component);
    expect(wrapper.text()).toEqual('Loading...');
  });
};

export const testErrorUI = (Component) => {
  it('should show error UI', async () => {
    const wrapper = mount(Component);
    await wait(0); // wait for response
    expect(wrapper.update().text()).toContain('Error');
  });
};

// THNX: http://blog.blakesimpson.co.uk/read/98-how-to-stub-date-in-jest-tests-building-a-stubdate-helper
export const stubDate = (fixedDate) => {
  let originalDate;

  beforeAll(() => {
    originalDate = Date;

    global.Date = class extends Date {
      constructor() {
        super();

        return fixedDate;
      }
    };
  });

  afterAll(() => {
    global.Date = originalDate;
  });
};
