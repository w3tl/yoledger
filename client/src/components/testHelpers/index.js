import React from 'react';
import { shallow, mount } from 'enzyme';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { makeExecutableSchema } from 'graphql-tools';
import typeDefs from './typeDefs';
import resolvers from './resolvers';

export const wait = (ms = 0) => new Promise(resolve => setTimeout(resolve, ms));

const executableSchema = makeExecutableSchema({
  typeDefs,
  resolvers,
});

const cache = new InMemoryCache();
const link = new SchemaLink({ schema: executableSchema });
export const client = new ApolloClient({
  link,
  cache,
});

export const withProvider = Component => props => (
  <ApolloProvider client={client}>
    <Component client={client} />
  </ApolloProvider>
);

export const testRenderWithoutError = (Component) => {
  it('renders without error', async () => {
    const wrapper = shallow(Component);
    await wait();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
};

export const testLoadingState = (Component) => {
  it('should render loading state initially', () => {
    const wrapper = mount(Component);
    expect(wrapper.text()).toEqual('Loading...');
  });
};

export const testErrorUI = (Component) => {
  it('should show error UI', async () => {
    const wrapper = mount(Component);
    await wait(0); // wait for response
    expect(wrapper.update().text()).toContain('Error');
  });
};

// THNX: http://blog.blakesimpson.co.uk/read/98-how-to-stub-date-in-jest-tests-building-a-stubdate-helper
export const stubDate = (fixedDate) => {
  const date = new Date(fixedDate);
  let originalDate;

  beforeAll(() => {
    originalDate = Date;

    global.Date = class extends Date {
      constructor() {
        super();

        return date;
      }
    };
  });

  afterAll(() => {
    global.Date = originalDate;
  });
};
