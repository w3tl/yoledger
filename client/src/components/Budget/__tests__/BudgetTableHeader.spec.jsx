import React from 'react';
import { mount } from 'enzyme';
import { BudgetTableHeaderComponent } from '../BudgetTableHeader';
import { periods } from '../mockData';
import { testRenderWithoutError } from '../../testHelpers';

describe('BudgetTableHeader', () => {
  testRenderWithoutError(<BudgetTableHeaderComponent periods={periods} onNext={() => {}} />);

  it('should handle next and prev button click', () => {
    const onNext = jest.fn();
    const output = mount(<BudgetTableHeaderComponent periods={periods} onNext={onNext} />);

    output.find('#nextButton').first().simulate('click');
    expect(onNext).toHaveBeenCalledWith(periods[1]);
  });
});
