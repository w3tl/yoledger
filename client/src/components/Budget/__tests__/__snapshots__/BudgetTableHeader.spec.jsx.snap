// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetTableHeader renders without error 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "flex": "1 100px",
      }
    }
  >
    Account
  </div>
  <div
    key="2018-06-10"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    10 Jun
  </div>
  <div
    key="2018-06-25"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    25 Jun
  </div>
  <div
    key="2018-07-10"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    10 Jul
  </div>
  <div
    key="2018-07-25"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    25 Jul
  </div>
  <div
    key="2018-08-10"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    10 Aug
  </div>
  <div
    key="2018-08-25"
    style={
      Object {
        "flex": 3,
      }
    }
  >
    25 Aug
  </div>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Button
      id="nextButton"
      onClick={[Function]}
      type="button"
    >
      Next
    </Button>
  </div>
</div>
`;
